lrlex/src/lib/ctbuilder.rs

```diff
@@ -463,29 +465,34 @@ where
         if let Some(inspect_lexerkind_cb) = self.inspect_lexerkind_cb {
             inspect_lexerkind_cb(lexerkind)?
         }
+        let header = Header::new();
         let (mut lexerdef, lex_flags): (Box<dyn LexerDef<LexerTypesT>>, LexFlags) = match lexerkind
         {
             LexerKind::LRNonStreamingLexer => {
-                let lexerdef = LRNonStreamingLexerDef::<LexerTypesT>::new_with_options(
-                    &lex_src,
-                    self.force_lex_flags.clone(),
-                    self.default_lex_flags.clone(),
-                )
-                .map_err(|errs| {
-                    errs.iter()
-                        .map(|e| {
-                            if let Some((line, column)) = line_cache.byte_to_line_num_and_col_num(
-                                &lex_src,
-                                e.spans().first().unwrap().start(),
-                            ) {
-                                format!("{} at line {line} column {column}", e)
-                            } else {
-                                format!("{}", e)
-                            }
-                        })
-                        .collect::<Vec<_>>()
-                        .join("\n")
-                })?;
+                let lexerdef =
+                    LRNonStreamingLexerDef::<LexerTypesT>::new_with_header(&lex_src, header)
```

Comment:
I didn't get to the trailhead before the defect in this patch came to me,
We aren't using `mark_used` on the `header`, but I'm not even sure we can, passing it by value.
`merge_from` isn't preserving `mark_used`, it's kind of assuming that all of that will happen *after* merging.

It isn't exactly clear to me whether we can pass this to `LRNonStreamingLexerDef` not as an owned value, but as a `&mut` ref.

Reply:
I haven't totally understood the details, but on the face of it (2) does sound like the better route to me. Is this PR the right place to do it or ...?

Reply:
I don't think this PR is the right place for it, it actually ends up undoing a lot of what is in this PR.

I do think (1) can go in this PR, it is basically the same approach taken here, I think maybe a good idea is to try the first approach, finishing it off in this PR if only to give us a fallback option that we can hold off on merging.  Just in case something goes awry with (2) which I also think seems preferable but also has some complexity trade-offs.

Reply:
Works for me.

Reply:
the first approach is implemented in this PR in [7468581](https://github.com/softdevteam/grmtools/pull/553/commits/74685813d647cc992ce488a339c84ad4a8227966) this didn't require trait modifications at all, and was actually a relatively minor change to the patch.

I'll try and start work on a new branch implementing second approach tomorrow.